lib/hijack/http.js

```diff
@@ -1,29 +1,82 @@
+/* eslint-disable prefer-rest-params */
+import { haveAsyncCallback } from '../utils';
+
 if (Package['http']) {
   const HTTP = Package['http'].HTTP;
-
-  var originalCall = HTTP.call;
+  const library = 'meteor/http';
+  const originalCall = HTTP.call;
 
   HTTP.call = function (method, url) {
-    var kadiraInfo = Kadira._getInfo();
-    if (kadiraInfo) {
-      var eventId = Kadira.tracer.event(kadiraInfo.trace, 'http', { method: method, url: url });
+    const tracer = Kadira.tracer;
+    const kadiraInfo = Kadira._getInfo();
+
+    const event = kadiraInfo ? tracer.event(kadiraInfo.trace, 'http', {
+      method,
+      url,
+      library,
+    }) : null;
+
+    if (!event) {
+      return originalCall.apply(this, arguments);
+    }
+
+    try {
+      const response = originalCall.apply(this, arguments);
+
+      // If the user supplied an asyncCallback,
+      // we don't have a response object and it handled asynchronously.
+      // We need to track it down to prevent issues like: #3
+      const endOptions = haveAsyncCallback(arguments) ? { async: true, library } : { statusCode: response.statusCode, library };
+
+      tracer.eventEnd(kadiraInfo.trace, event, endOptions);
+
+      return response;
+    } catch (ex) {
+      tracer.eventEnd(kadiraInfo.trace, event, { err: ex.message });
+
+      throw ex;
+    }
+  };
+}
+
+if (Package['fetch']) {
+  const library = 'meteor/fetch';
+  const originalCall = Package['fetch'].fetch;
+  const Request = Package['fetch'].Request;
+
+  Package['fetch'].fetch = function (url, opts) {
```

Comment:
As mentioned in meteor/fetch:

https://github.com/meteor/meteor/tree/devel/packages/fetch

> In Node, the [node-fetch](https://www.npmjs.com/package/node-fetch) polyfill is used. Note: unlike the client polyfills, the Node polyfill does not define the fetch function globally. However, any application or package that depends on the Meteor fetch package can refer to fetch as if it was a global function (or import { fetch } from "meteor/fetch").

I have no ideas how to override global fetch

Reply:
Meteor takes `Package['fetch'].fetch` and creates a global variable for it. How you are wrapping the fetch package should work, it just will only be wrapped for packages that load after it is wrapped.

Reply:
I just tried this:

`fetch = Package['fetch'] = function ...` and it works. But `fetch` isn't a function ,it is a `module`:

```
I20220303-03:13:45.681(3)? <ref *1> [Function: fetch] {
I20220303-03:13:45.681(3)?   isRedirect: [Function (anonymous)],
I20220303-03:13:45.681(3)?   Promise: [Function: Promise] {
I20220303-03:13:45.681(3)?     Fiber: [Function: Fiber] {
I20220303-03:13:45.681(3)?       yield: [Function (anonymous)],
I20220303-03:13:45.681(3)?       current: [Fiber],
I20220303-03:13:45.681(3)?       poolSize: 120,
I20220303-03:13:45.681(3)?       fibersCreated: 12
I20220303-03:13:45.682(3)?     },
I20220303-03:13:45.682(3)?     awaitAll: [Function (anonymous)],
I20220303-03:13:45.682(3)?     await: [Function (anonymous)],
I20220303-03:13:45.682(3)?     async: [Function (anonymous)],
I20220303-03:13:45.682(3)?     asyncApply: [Function (anonymous)]
I20220303-03:13:45.682(3)?   },
I20220303-03:13:45.682(3)?   default: [Circular *1],
I20220303-03:13:45.682(3)?   Headers: [class Headers],
I20220303-03:13:45.682(3)?   Request: [class Request],
I20220303-03:13:45.682(3)?   Response: [class Response],
I20220303-03:13:45.682(3)?   FetchError: [Function: FetchError]
I20220303-03:13:45.682(3)? }
```

Is it good idea to override it?